adoc/extensions/sycl_khr_command_graph.adoc

```diff
@@ -252,111 +253,423 @@ a@ Constructs a [code]#depends_on_all_leaves# property instance.
+<<table.command_graph.graph_state>>.
+
+[[table.command_graph.graph_state]]
+.Values of the [code]#graph_state# enum.
+[width="100%",options="header",separator="@",cols="30%,70%"]
+|====
+a@ Value a@ Description
 
-[source, c++]
+a@
+[source]
+----
+graph_state::modifiable
 ----
-namespace sycl::khr {
-// State of a graph
-enum class graph_state {
-  modifiable,
-  executable
-};
 
-// New object representing graph
-template<graph_state State = graph_state::modifiable>
-class command_graph {};
+a@ The [code]#command_graph# is under construction and new nodes or edges may
+be added to it at any time. Graphs are created in this state and cannot be executed until
+they have been finalized into the [code]#executable# state.
 
-template<>
-class command_graph<graph_state::modifiable> {
-public:
-  command_graph(const context& syclContext, const device& syclDevice,
-                const property_list& propList = {});
+This is the default template argument for the [code]#command_graph# class.
 
-  command_graph(const queue& syclQueue,
-                const property_list& propList = {});
+a@
+[source]
+----
+graph_state::executable
+----
 
-  command_graph<graph_state::executable>
-  finalize(const property_list& propList = {}) const;
+a@ A [code]#command_graph# in this state is a fixed snapshot of the state and
+topology of the modifiable graph it was finalized from. Subsequent changes to
+that modifiable graph have no effect on the executable graph(s) created from
+it.
+A graph in this state is ready for execution, but can no longer have new nodes
+or edges added to it.
+|====
 
-  void begin_recording(queue& recordingQueue, const property_list& propList = {});
-  void begin_recording(const std::vector<queue>& recordingQueues, const property_list& propList = {});
+=== Command-graph class interface
 
-  void end_recording();
-  void end_recording(queue& recordingQueue);
-  void end_recording(const std::vector<queue>& recordingQueues);
+[source,,linenums]
+----
+include::{header_dir}/extensions/khr_graph/command_graph.h[lines=4..-1]
+----
 
-  node add(const property_list& propList = {});
+[[subsec:khr-graph-command-graph-ctors]]
+=== Constructors
 
-  template<typename T>
-  node add(T cgf, const property_list& propList = {});
+All [code]#command_graph# constructors take a parameter named [code]#propList#
+which allows the application to pass zero or more properties.
+These properties may specify additional effects of the constructor and resulting
+[code]#command_graph# object.
+See <<subsec:khr-graph-command-graph-properties>> for the [code]#command_graph#
+properties that are defined by this extension.
 
-  node add(dynamic_command_group& dynamicCG, const property_list& propList = {});
+.[apititle]#Context/device constructor#
+[source, role=synopsis,id=api:khr-command-graph-ctx-dev-ctor]
+----
+command_graph(const context& syclContext, const device& syclDevice,
+                const property_list& propList = {})
```

Comment:
```suggestion
command_graph(const context& syclContext, const device& syclDevice,
              const property_list& propList = {})
```
Nipicking that it would be nice for the params to line up in rendered view. Right now
![image](https://github.com/user-attachments/assets/82047368-5ebb-48b8-a03b-bcc0603cc9dc)